### ISSUE TITLE
Option to allow page to load when its a 404 error

### ISSUE THREAD

**Issue by RileyEv** [miguel-perez/smoothState.js#293]
Would it be possible to add the option to continue to load even if its a 404 error? I have custom error pages that I would also like to use with smoothSate.

**Comment by CHEWX** [comment]
That sounds like a routing issue rather than Smoothstate.

**Comment by jamie-jenks** [comment]
+1

**Comment by treechime** [comment]
+1

**Comment by treechime** [comment]
Actually, just found that removing the debug option seems to fix this.

**Comment by rulatir** [comment]
@CHEWX Could you elaborate? How do you imagine a routing-based **solution** to this problem? You do realize that it is smoothState that is doing the ajax requests, and the 404 condition is only known to have occurred **after** smoothState receives the response? Where/when does a client-side router plug into this? Before you click on a bad link, it is too early. Next time smoothState grants you the opportunity to alter its behavior, it is already too late.

**Comment by CHEWX** [comment]
@rulatir Ok putting aside the logic, so you are saying if the OP turned off smoothState and clicked a dead link then it routes correctly to the custom 404, yet if they turn smoothState back on, it doesn't route correctly? I can't see that being the case. Unless I am missing something here...

If you purposely change a link in the DOM, smoothState routes correctly to a 404 on all my websites.

**Comment by rulatir** [comment]
@CHEWX Do you use a "pseudo 404" page, i.e. one that says "We're sorry!", shows site map / search box / whatever **but is served with HTTP response code 200 rather than 404**? If that is the case then of course it works, because it's just 200 to smoothState and to the underlying $.ajax(). But if the server serves that page with real `HTTP/1.1 404 Not Found`, then it definitely breaks smoothState, at least for me.

**Comment by rulatir** [comment]
What I am saying is that when the $.ajax() request issued by smoothState receives a response with status code 404, then smoothState treats it like an error and throws an exception.

**Comment by CHEWX** [comment]
@rulatir Yes exactly, so like I said, it's a routing issue, not smoothState. If you were to have a custom 404, then it would have to have a 200 response code, because you cannot style a 404 response.

**Comment by CHEWX** [comment]
Adding this functionality to smoothState, isn't what smoothState is for. Plugins like this should handle a single job. Not routing issues. Because the way I see it, if you weren't using smoothState, then you'd have to handle the 404 response. So why when using smoothState should you let the plugin handle this route.

It's a pretty simple option to add in though. Although I don't personally think this plugin should handle such a use-case.

**Comment by rulatir** [comment]
@CHEWX no, you are wrong, a custom 404 page should be served with 404 code, it's a best practice that smoothState should handle. The HTTP standard defines some error responses as displayable to the user, and serving a 400+ response code with a HTML document in the response body is **the** right way to do it.

A simply stated requirement for smoothState is that it shouldn't break websites. If I don't use smoothState and I **can see** the HTML document served in the response body of a 404 response, then I **should be able to see** the same with smoothState.
